docs: describe the purpose of the _dev dirs (#954)
elastic/integrations: docs/developer_workflow_design_build_test_integration.md

# Developer workflow: build and test the integration

> ## Directory layout for tests

Reviewer: Let's rename it to: Development Extension ("_dev" directories)

> ## Directory layout for tests

Author: ```suggestion
## Development Extensions: '_dev' directories
```

> The `_dev` folder is an extension to [the standard package spec](https://github.com/elastic/package-spec), which has been created for development purposes. Currently, there are two _dev directories:

Reviewer: > Currently, there are two _dev directories:

Maybe rephrase it: The `_dev` directories can be defined on the following levels:

> The `_dev` folder is an extension to [the standard package spec](https://github.com/elastic/package-spec), which has been created for development purposes. Currently, there are two _dev directories:

Author: ```suggestion
The `_dev` directory is an extension to [the standard package spec](https://github.com/elastic/package-spec), which has been created for development purposes. It can be defined on the following levels:
```

> 1. **Package** level (https://github.com/elastic/package-spec/tree/master/versions/1/_dev) - it contains resources to "build" docs content and "deploy" services for system tests.

Reviewer: Would you mind replacing the `_dev` urls with spec hyperlinks: https://github.com/elastic/package-spec/blob/master/versions/1/data_stream/_dev/spec.yml ?

This way the user will jump the exact place where these resources are defined. It may be less confusing.

> 1. **Data-stream** level (https://github.com/elastic/package-spec/tree/master/versions/1/data_stream/_dev) - it contains "test" definitions for pipeline and system tests.

Reviewer: definitions for, i.e. pipeline and system tests.

Ok, let me explain:
We have also asset and static tests. They don't require config files, but configs can be used to mark them as optional.

> 1. **Data-stream** level (https://github.com/elastic/package-spec/tree/master/versions/1/data_stream/_dev) - it contains "test" definitions for pipeline and system tests.

Reviewer: Sorry I wasn't clear enough. I suggested to replace the URLs with hyperlinks:

[spec.yml](https://github.com/elastic/package-spec/tree/master/versions/1/_dev/spec.yml) - this form would be much shorter.

> 1. **Data-stream** level (https://github.com/elastic/package-spec/tree/master/versions/1/data_stream/_dev) - it contains "test" definitions for pipeline and system tests.

Reviewer: **Package** level ([spec.yml](https://github.com/elastic/package-spec/tree/master/versions/1/_dev/spec.yml)):  the directory contains ...

# Developer workflow: build and test the integration

> ## Development Extensions: '_dev' directories

Reviewer: As I suggested [here](https://github.com/elastic/integrations/pull/954#pullrequestreview-646772651) - it would be more consistent to put it under `definitions.md`.

Author: Ahh sorry, I missed that comment :/ I'll push a change with your suggestion
